### PR TITLE
fix: add check install step for nfs

### PR DESCRIPTION
**Pull request by Metrora**
Signed-off-by: Liucw <liu.chuwei@99cloud.net>

### What type of PR is this?
/kind feature

### Which issue(s) this PR fixes:

Fixes https://github.com/kubeclipper-labs/kubeclipper/issues/238

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note
None
```

### Additional documentation, usage docs, etc.:

```docs

```
@x893675

### CODE REVIEW

File: pkg/component/common/check_csi_install.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+)
+
+func init() {
+	if err := component.RegisterAgentStep(fmt.Sprintf(component.RegisterStepKeyFormat, ciName, version, CheckCSIInstall), &CheckInstall{}); err != nil {
+		panic(err)
+	}
+}
+
+var (
+	_ component.StepRunnable = (*CheckInstall)(nil)
```
Comment (x893675) [inline pkg/component/common/check_csi_install.go:27]: rename CSIHealthCheck

File: pkg/component/common/check_csi_install.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+)
+
+const (
+	ciName           = "check-install"
```
Comment (x893675) [inline pkg/component/common/check_csi_install.go:31]: ciName change to csiHealthCheck, variable name should be human readable

File: pkg/component/common/check_csi_install.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	ManifestsDir     = "/tmp"
+	CSITestFile      = "csi-test.yaml"
+	CheckCSIInstall  = "CheckCSIInstall"
```
Comment (x893675) [inline pkg/component/common/check_csi_install.go:34]: rename all CSIInstall to healthcheck

File: pkg/component/common/check_csi_install.go [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+  storageClassName: {{.StorageClassName}}
+`
+)
+
+type CheckInstall struct {
+	CsiType string
```
Comment (x893675) [inline pkg/component/common/check_csi_install.go:51]: StorageClassName

File: pkg/component/common/check_csi_install.go [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+}
+
+func (c *CheckInstall) Install(ctx context.Context, opts component.Options) ([]byte, error) {
+	// check install csi success or not
```
Comment (x893675) [inline pkg/component/common/check_csi_install.go:55]: apply pvc yaml first

File: pkg/component/nfs/nfs.go [hunk 6]
```diff
@@ -264,10 +264,20 @@ func (n *NFSProvisioner) InitSteps(ctx context.Context) error {
 					Type:         v1.CommandShell,
 					ShellCommand: []string{"kubectl", "apply", "-f", filepath.Join(n.ManifestsDir, fmt.Sprintf(filenameFormat, n.StorageClassName))},
 				},
+				{
+					Type:         v1.CommandShell,
+					ShellCommand: []string{"kubectl", "apply", "-f", filepath.Join(n.ManifestsDir, common.CSITestFile)},
```
Comment (x893675) [inline pkg/component/nfs/nfs.go:269]: remove

File: pkg/component/nfs/nfs.go [hunk 7]
```diff
@@ -458,6 +468,12 @@ func (n *NFSProvisioner) renderNameSpace(w io.Writer) error {
 	return err
 }
 
+func (n *NFSProvisioner) renderNFSTest(w io.Writer) error {
+	at := tmplutil.New()
+	_, err := at.RenderTo(w, common.CheckCSITemplate, n)
+	return err
```
Comment (x893675) [inline pkg/component/nfs/nfs.go:474]: remove

File: pkg/component/common/check_csi_install.go [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+	"github.com/kubeclipper/kubeclipper/pkg/component/utils"
+	"github.com/kubeclipper/kubeclipper/pkg/logger"
+	v1 "github.com/kubeclipper/kubeclipper/pkg/scheme/core/v1"
+	"github.com/kubeclipper/kubeclipper/pkg/utils/cmdutil"
+	"github.com/kubeclipper/kubeclipper/pkg/utils/strutil"
+)
+
+func init() {
+	c := &CSIHealthCheck{}
+	if err := component.RegisterAgentStep(fmt.Sprintf(component.RegisterStepKeyFormat, CsiHealthCheck, version, HealthCheck), &CSIHealthCheck{}); err != nil {
```
Comment (lixd) [inline pkg/component/common/check_csi_install.go:28]: to `	if err := component.RegisterAgentStep(fmt.Sprintf(component.RegisterStepKeyFormat, CsiHealthCheck, version, HealthCheck), c); err != nil {
`

File: pkg/component/common/check_csi_install.go [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+		panic(err)
+	}
+	if err := component.RegisterTemplate(fmt.Sprintf(component.RegisterTemplateKeyFormat, CsiHealthCheck, version, HealthCheck), c); err != nil {
+		panic(err)
+	}
+}
+
+var (
+	_ component.StepRunnable = (*CSIHealthCheck)(nil)
```
Comment (lixd) [inline pkg/component/common/check_csi_install.go:37]: add `	_ component.TemplateRender = (*CSIHealthCheck)(nil)
`

File: pkg/component/common/check_csi_install.go [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+		logger.Warnf("kubectl apply %s failed: %s", checkCSIHealthFile, err.Error())
+	}
+	// check install csi success or not
+	if err := utils.RetryFunc(ctx, opts, 10*time.Second, "checkCSIInstall", c.checkCSIHealth); err != nil {
+		return nil, err
+	}
+	// delete pod, pvc, pv
+	_, err = cmdutil.RunCmdWithContext(ctx, opts.DryRun, "kubectl", "delete", "-f", filePath)
+	if err != nil {
+		logger.Warnf("kubectl delete %s failed: %s", checkCSIHealthFile, err.Error())
```
Comment (lixd) [inline pkg/component/common/check_csi_install.go:78]: to
```go
		logger.Warnf("kubectl delete %s failed: %s", filePath, err.Error())

```

File: pkg/component/common/check_csi_install.go [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+	return nil, nil
+}
+
+func (c *CSIHealthCheck) NewInstance() component.ObjectMeta {
+	return &CSIHealthCheck{}
+}
+
+// GetCheckCSIHealthStep get the common step
+func (c *CSIHealthCheck) GetCheckCSIHealthStep(nodes []v1.StepNode, storageClassName string) ([]v1.Step, error) {
+	steps := make([]v1.Step, 0)
```
Comment (lixd) [inline pkg/component/common/check_csi_install.go:108]:
```go

	steps := make([]v1.Step, 0, 2)
	c.StorageClassName = storageClassName
	aData, err := json.Marshal(c)
	if err != nil {
		return steps, err
	}
```

File: pkg/component/common/check_csi_install.go [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+	_, err := at.RenderTo(w, CSIHealthCheckTemplate, c)
+	return err
+}
+
+func (c *CSIHealthCheck) Render(ctx context.Context, opts component.Options) error {
+	if err := os.MkdirAll(ManifestsDir, 0755); err != nil {
+		return err
+	}
+	filePath := filepath.Join(ManifestsDir, checkCSIHealthFile)
+	err := fileutil.WriteFileWithContext(ctx, filePath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644,
```
Comment (lixd) [inline pkg/component/common/check_csi_install.go:167]: return directly

```go
return fileutil.WriteFileWithContext(ctx, filePath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644,
		c.renderCSIHealthCheckPVC, opts.DryRun)

```

**Review by lixd — APPROVED** [review]
/lgtm

**Pull request merged**